packages/menu/src/Menu.tsx

```diff
@@ -225,13 +224,16 @@ function Menu({
     }
   }, [currentSubMenu]);
 
-  const updatedChildren = React.useMemo(() => updateChildren(children), [
-    children,
-    focused,
-    currentSubMenu,
-    open,
-    refs,
-  ]);
+  const updatedChildren = React.useMemo(() => {
+    if (
+      children == null ||
+      ['boolean', 'number', 'string'].includes(typeof children)
+    ) {
+      return;
```

Comment:
We're just saying you don't have to update the children if the typeof children isn't a ReactElement. I was doing some weird hacky thing with TS earlier to get it to work very specifically in context, but it was causing problems when trying to use Menu as it should work in other places. This removes those TS errors. For example, I was trying to conditionally render a MenuItem but was getting an error from TS because previously we said the children of Menu had to be an array of React Elements and undefined/null is not included in that